pulp_2_tests/tests/platform/api_v2/test_consumer.py

```diff
@@ -67,3 +70,70 @@ def test_all(self):
             self.assertEqual(result['consumer_id'], consumer['consumer']['id'])
             self.assertEqual(result['distributor_id'], body['distributor_id'])
             self.assertEqual(result['repo_id'], body['repo_id'])
+
+
+class RegisterAndUpdateConsumerTestCase(unittest.TestCase):
+    """Show that one can `register and update a consumer`_.
+
+        The call to `consumer register api` should return a x.509 certificate,
+        that should be useful in updating a consumer and for other actions.
+
+    .. _register and update a consumer:
+        https://docs.pulpproject.org/dev-guide/integration/rest-api/consumer/cud.html#register-a-consumer
+    """
+
+    def test_all(self):
+        """Register and Update a consumer.
+
+        Do the following:
+
+        1. Register a consumer with the consumer API
+        2. Save the certificate returned in the response
+        3. Update the same consumer with new details by using the client certificates
+
+        Assert that:
+
+        * The response for registering a consumer has a x.509 certificate.
+        * The response for update with certificate has HTTP 200 .
+        * The response body contains the correct values.
+
+        Refer:
+
+        * `Pulp Smash #1007 <https://github.com/PulpQE/pulp-smash/issues/1007>`_
+        """
+        cfg = config.get_config()
+        client = api.Client(cfg)
+
+        # Step 1
+        consumer = client.post(CONSUMERS_PATH, {'id': utils.uuid4()}).json()
+        self.addCleanup(client.delete, consumer['consumer']['_href'])
+        with self.subTest(comment='check register call returns with certificate'):
+            self.assertIn('certificate', consumer)
```

Comment:
I see this code below:

```python
certificate = consumer['certificate']
```

So why is this test continuing even when "certificate" isn't in the "consumer" dict? Drop the surrounding `subTest`.